client/src/modules/list/model/reducer.js

```diff
@@ -199,6 +200,24 @@ const lists = (state = {}, action) => {
     }
     case CohortActionTypes.ARCHIVE_SUCCESS:
       return {};
+    case CohortActionTypes.LEAVE_SUCCESS: {
+      const { payload: leavedCohortId } = action;
+
+      return _keyBy(
+        _filter(state, list => {
+          const { cohortId, isMember, type } = list;
+          if (
+            !cohortId ||
+            cohortId !== leavedCohortId ||
+            type === ListType.LIMITED ||
+            (type === ListType.SHARED && isMember)
+          ) {
+            return list;
+          }
+        }),
+        '_id'
+      );
+    }
```

Comment:
can you explain what this reducer is doing?
I can't see any leaving process from members, owners array for the user here.

Reply:
If the user leaves the cohort we have to remove some lists from the store, lists to which the user lost access to by leaving this cohort. (cohort's shared lists where they were a viewer only). And this is what this reducer does - filters the lists to remove those ones which are forbidden for this user now.